addons/product/models/product_attribute.py

```diff
@@ -26,6 +26,20 @@ class ProductAttribute(models.Model):
         string="Create Variants",
         help="Check this if you want to create multiple variants for this attribute.", required=True)
 
+    @api.constrains('name')
+    def _check_name(self):
+        """ The name must be unique """
+        for att in self:
+            domain = [('name', '=', att.name), ('id', '!=', att.id)]
+            if self.search(domain):
+                raise ValidationError(_('The name be unique!'))
```

Comment:
```suggestion
                raise ValidationError(_('The name must be unique!'))
```

No TODO to be put this as SQL unique constraint?